source/dynamical_systems/include/dynamical_systems/Linear.hpp

```diff
@@ -185,9 +186,9 @@ inline void Linear<state_representation::CartesianState>::set_base_frame(const s
     throw state_representation::exceptions::EmptyStateException(base_frame.get_name() + " state is empty");
   }
   DynamicalSystem<state_representation::CartesianState>::set_base_frame(base_frame);
-  if (!this->get_attractor().is_empty()) {
```

Comment:
The previous syntax was actually better, since it doesn't require the intermediate variable in the case that the attractor is empty. Granted, the intermediate variable will be a const reference and so barely cost anything, and the compiler will almost certainly optimise it away anyway.